internal/mongodoc/resource_test.go

```diff
@@ -0,0 +1,369 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package mongodoc_test
+
+import (
+	"time"
+
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+	"gopkg.in/juju/charm.v6-unstable"
+	"gopkg.in/juju/charmrepo.v2-unstable/csclient/params"
+	"gopkg.in/mgo.v2/bson"
+
+	"gopkg.in/juju/charmstore.v5-unstable/internal/mongodoc"
+)
+
+const fingerprint = "0123456789abcdef0123456789abcdef0123456789abcdef"
+
+var (
+	_ = gc.Suite(&ResourceSuite{})
+	_ = gc.Suite(&ResourcesSuite{})
+)
+
+type ResourceSuite struct{}
+
+func (s *ResourceSuite) TestNewResourceQueryWithRevision(c *gc.C) {
+	cURL := charm.MustParseURL("cs:trusty/spam-2")
+
+	query := mongodoc.NewResourceQuery(cURL, "eggs", 3)
+
+	c.Check(query, jc.DeepEquals, bson.D{
```

Comment:
This really doesn't test anything worth testing. Without testing this against an actual mongo database, this doesn't give us any assurance at all that the code will really work - we're just testing that what we return is what we return, not that it will actually work as the expected query.

More reason for the resource query and associated tests to live in the internal/charmstore package.